tztrout/data.py

```diff
@@ -26,6 +26,67 @@
 AU_AREA_CODE_TO_STATE_MAP_PATH =  os.path.join(basepath, 'data/au_area_code_to_state.json')
 
 
+def deduplicator():
+    """Deduplicate strings in memory using a canonical mapping.
+
+    Works similarly to intern() but supports unicode as well.
+
+    >>> a = chr(1) + chr(1)  # Create two strings
+    >>> b = chr(1) + chr(1)
+
+    >>> a == b  # These have the same contents
+    True
+    >>> id(a) == id(b)  # But they're different objects
+    False
+
+    >>> deduplicate = deduplicator()  # Deduplicating them returns the same object
+    >>> id(deduplicate(a)) == id(deduplicate(b))
+    True
+    """
+    memo = {}
+
+    def deduplicate(item):
+        if item in memo:
+            return memo[item]
+        memo[item] = item
+        return item
+
+    return deduplicate
+
+
+class InMemoryZipData(object):
```

Comment:
This class could use a docstring which, besides explaining what this does and how to use it, also notes that this is expensive to instantiate (both in terms of memory and time). On my MBP this takes ~350ms:
```
In [4]: %time InMemoryZipData()
CPU times: user 319 ms, sys: 27.9 ms, total: 347 ms
Wall time: 347 ms
Out[4]: <tztrout.data.InMemoryZipData at 0x105c64490>
```